### DISCUSSION TITLE
Hint int property is a Variant type enumeration

### DISCUSSION THREAD

**Discussion by timothyqiu** [godotengine/godot-proposals#5857] [Engine Core]
Currently no way to specify a `int` export variable is any of `TYPE_INT`, `TYPE_ARRAY`, `TYPE_AABB`, etc.

Editor code (e.g. AnimationTrackKeyEdit and TileSet) hints the variable is an enum, and populates the enum hint with the corresponding Variant type names. This tedious to do in GDScript as `Variant::get_type_name()` is not exposed.

I think there're two ways to implement this:

1. Expose `Variant::get_type_name()` so the user can fill enum hint easily.
2. Add a `PROPERTY_HINT_INT_IS_VARIANT_TYPE` hint.

**Comment by dalexeev** [comment]
1. Now there is `type_string()`. See godotengine/godot#69624.
2. It's `PropertyInfo(Variant::INT, "prop_name", PROPERTY_HINT_NONE, "", PROPERTY_USAGE_CLASS_IS_ENUM, "Variant.Type")`.

In GDScript you can also use `@export var type: Variant.Type`:
![](https://github.com/user-attachments/assets/3efa73fd-0262-4393-b5e7-39012b180f3d)
